### PR TITLE
update the POD for PGauxiliaryFunctions.pl

### PR DESCRIPTION
**Pull request by pstaabp**
This updates the POD for PGauxiliaryFunctions.pl, which was not in the correct format to be able to be parsed correctly. 

To test run `WEBWORK_ROOT/bin/dev_scripts/generate-ww-pg-pod.pl` to see the output in html format.

### CODE REVIEW

**Review by drgrice1 — APPROVED** [review]
Looks good.  This is much better than just fixing bad formatting the POD in this file previously had.  This actually makes it useful.

File: macros/PGauxiliaryFunctions.pl [hunk 1]
```diff
@@ -1,65 +1,129 @@
...
+
+}
+
+=head3 step function
+
+=pod
+
+	Usage: step(x);
+
+returns the step function (or Heavyside function) with jump at x=0.  That is when x<0, it returns 0, when x>=0 the function returns 1.
```
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:58]: nit: "Heaviside" -- named for Oliver Heaviside... (another instance on L66)

File: macros/PGauxiliaryFunctions.pl [hunk 2]
```diff
@@ -143,32 +273,60 @@ sub gcf {
 	return gcf($b,@_);
 }
 
-# greatest common factor.
-# same as gcf, but both names are sufficiently common names
-# ^function gcd
-# ^uses gcf
+=head3 gcf function
```
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:276]: this is for `gcd`, not `gcf`

File: macros/PGauxiliaryFunctions.pl [hunk 3]
```diff
@@ -143,32 +273,60 @@ sub gcf {
...
-# same as gcf, but both names are sufficiently common names
-# ^function gcd
-# ^uses gcf
+=head3 gcf function 
+
+=pod
+
+	Usage: gcd(@arr);
+
+returns the greatest common denominator of the array @arr of integers.
```
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:282]: greatest common _divisor_ (not denominator)

File: macros/PGauxiliaryFunctions.pl [hunk 4]
```diff
@@ -270,9 +443,24 @@ sub random_pairwise_coprime {
...
+
+=pod
+
+	Usage: isPrime(n);
+
+returns 1 if n is prime and 0 otherwise. 
+
+Example: 
+	
+	gcf(20,30,45) returns 5.
```
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:456]: not an example of using `isPrime`

File: macros/PGauxiliaryFunctions.pl [hunk 5]
```diff
@@ -302,13 +501,21 @@ sub reduce {
 	@frac;
 }
 
+=head3 preFormat function 
+
+=pod
+
+	Usage: preFormat($scalar,"quoted string");
+
+returns the string preformated with the $scalar as 0,1, or -1  takes a number and fixed object, as in "$a x" and formats
```
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:510]: pre-formatted (two 't's)

**Review by drdrew42 — COMMENTED** [review]
Just a bunch of tiny edits...

File: macros/PGauxiliaryFunctions.pl (hunk 1 above)
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:58]:
```suggestion
returns the step function (or Heaviside function) with jump at x=0.  That is when x<0, it returns 0, when x>=0 the function returns 1. 
```

File: macros/PGauxiliaryFunctions.pl [hunk 6]
```diff
@@ -1,65 +1,129 @@
...
+returns the step function (or Heavyside function) with jump at x=0.  That is when x<0, it returns 0, when x>=0 the function returns 1. 
+
+Example:
+
+	step(3.14159) returns 1
 
 =cut
 
-# ^function step
 sub step {     # heavyside function (1 or x>0)
```
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:66]:
```suggestion
sub step {     # heaviside function (1 or x>0)
```

File: macros/PGauxiliaryFunctions.pl (hunk 2 above)
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:276]:
```suggestion
=head3 gcd function 
```

File: macros/PGauxiliaryFunctions.pl (hunk 3 above)
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:282]:
```suggestion
returns the greatest common divisor of the array @arr of integers.  
```

File: macros/PGauxiliaryFunctions.pl (hunk 4 above)
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:456]:
```suggestion
	isPrime(1337) returns 0.
```

File: macros/PGauxiliaryFunctions.pl (hunk 5 above)
Comment (drdrew42) [inline macros/PGauxiliaryFunctions.pl:510]:
```suggestion
returns the string pre-formatted with the $scalar as 0,1, or -1  takes a number and fixed object, as in "$a x" and formats
```

**Review by drdrew42 — CHANGES_REQUESTED** [review]
I learned a ~~new~~ [github thing](https://haacked.com/archive/2019/06/03/suggested-changes/).

**Comment by drgrice1** [comment]
Nice.  That is a useful feature.

**Comment by pstaabp** [comment]
Not sure I'm using GH efficiently, but I think I made all of suggested changes.  Thanks @drdrew42 .

**Review by drdrew42 — APPROVED** [review]

**Pull request merged**